tools/Test/SmokeTest/PrepareRequiredPowershell.ps1

```diff
@@ -5,6 +5,95 @@ param(
   $requiredPsVersion
 )
 
+$IsLinuxEnv = (Get-Variable -Name "IsLinux" -ErrorAction Ignore) -and $IsLinux
+$IsMacOSEnv = (Get-Variable -Name "IsMacOS" -ErrorAction Ignore) -and $IsMacOS
+$IsWinEnv = !$IsLinuxEnv -and !$IsMacOSEnv
+
+# The PowerShell destination
+if (-not $Destination){
+    if ($IsWinEnv) {
+        $Destination = "D:\a\_work\1\s"
+    }elseif($IsLinuxEnv){
+        $Destination = "/mnt/vss/_work/1/s"
+    }elseif($IsMacOSEnv){
+        $Destination = "/Users/runner/work/1/s"
+    }
+}
+
+# Preview version package download path
+$TempDir = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+function Expand-ArchiveInternal {
+    [CmdletBinding()]
+    param(
+        $Path,
+        $DestinationPath
+    )
+
+    if((Get-Command -Name Expand-Archive -ErrorAction Ignore))
+    {
+        Expand-Archive -Path $Path -DestinationPath $DestinationPath
+    }
+    else
+    {
+        Add-Type -AssemblyName System.IO.Compression.FileSystem
+        $resolvedPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($Path)
+        $resolvedDestinationPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($DestinationPath)
+        [System.IO.Compression.ZipFile]::ExtractToDirectory($resolvedPath,$resolvedDestinationPath)
+    }
+}
+
+function Install-Preview-PowerShell {
+  if (-not $IsWinEnv) {
+    $architecture = "x64"
```

Comment:
Please use [System.Environment]::Is64BitOperatingSystem to check the bit size of architecture.